pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -649,13 +658,42 @@ func (a *index) inRevision(obj any) bool {
 
 // All return all known workloads. Result is un-ordered
 func (a *index) All() []model.AddressInfo {
-	res := slices.Map(a.workloads.List(), modelWorkloadToAddressInfo)
+	var res []model.AddressInfo
+	// Add all workloads
+	for _, wl := range a.workloads.List() {
+		// If EnableAmbientMultiNetwork is enabled, we only want to add workloads that are local to this cluster
+		// Non cluster local workloads will be added by the service lookup
+		if features.EnableAmbientMultiNetwork && wl.Workload.ClusterId != string(a.ClusterID) {
```

Comment:
This breaks single network multicluster; clusterID isn't really sufficient here for exclusion. We really want to know the network of this workload right?

Reply:
Agree - will fix and this will be removed with the split horizon changes.

Reply:
Is `clusterID` sufficient for now since we are assuming for alpha each control plane is in one network?